kpops/components/common/kubernetes_model.py

```diff
@@ -1,12 +1,87 @@
 import enum
+from collections.abc import Iterator
+from typing import Any
 
-from pydantic import Field
+import yaml
+from pydantic import ConfigDict, Field
+from typing_extensions import override
 
 from kpops.utils.docstring import describe_attr
 from kpops.utils.pydantic import DescConfigModel
 
 # Matches plain integer or numbers with valid suffixes: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory
 MEMORY_PATTERN = r"^\d+([EPTGMk]|Ei|Pi|Ti|Gi|Mi|Ki)?$"
+K8S_LABEL_MAX_LEN = 63
+
+
+class ManagedFieldsEntry(DescConfigModel):
+    # Define this class based on its actual structure
+    fields: dict[str, Any] | None = None
+
+
+class OwnerReference(DescConfigModel):
+    # Define this class based on its actual structure
+    apiVersion: str
+    kind: str
+    name: str
+    uid: str
+    controller: bool | None = None
+    blockOwnerDeletion: bool | None = None
+
+
+class ObjectMeta(DescConfigModel):
+    """Metadata for all Kubernetes objects."""
+
+    annotations: dict[str, str] | None = None
+    creationTimestamp: str | None = Field(
+        None, description="Timestamp in RFC3339 format"
+    )
+    deletionGracePeriodSeconds: int | None = None
+    deletionTimestamp: str | None = Field(
+        None, description="Timestamp in RFC3339 format"
+    )
+    finalizers: list[str] | None = None
+    generateName: str | None = None
+    generation: int | None = None
+    labels: dict[str, str] | None = None
+    managedFields: list[ManagedFieldsEntry] | None = None
+    name: str | None = None
+    namespace: str | None = None
+    ownerReferences: list[OwnerReference] | None = None
+    resourceVersion: str | None = None
+    selfLink: str | None = Field(
+        None,
+        description="Deprecated field, not populated by Kubernetes in modern versions",
+    )
+    uid: str | None = None
+    model_config = ConfigDict(extra="allow")
+
+
+class KubernetesManifest(DescConfigModel):
+    # TODO: this is not working without the 'alias'
+    api_version: str = Field(alias="apiVersion")
+    kind: str
+    metadata: ObjectMeta
+
+    model_config = ConfigDict(extra="allow")
+
+    @classmethod
+    def from_yaml(
+        cls, /, content: str
+    ) -> Iterator["KubernetesManifest"]:  # TODO: typing.Self for Python 3.11+
+        manifests: Iterator[dict[str, Any]] = yaml.load_all(content, yaml.Loader)
+        for manifest in manifests:
+            yield cls(**manifest)
+
+    @override
+    def model_dump(self, **_: Any) -> dict[str, Any]:
```

Comment:
why do we need this override?

Reply:
I found it better to override this and enforce the exclusion settings in the base class instead of doing it separately everytime we call the method.